### DISCUSSION TITLE
Add support of $timestamp$ (or $epoch$) for "trusted scriptlets"

### DISCUSSION THREAD

**Discussion by K-mikaZ** [uBlockOrigin/uBlock-issues#3296] [Ideas]
Due to the fact that the `trusted-click-element` item button works for close cookie banner but  the page stays frozen after on some sites (like "ileauxepices.com")

Some sites use timestamp to set settings (e.g. cookies) such as:
`ileauxepices.com##+js(trusted-set-cookie, ets_am_customer_reward_tab, undefined)`
`ileauxepices.com##+js(trusted-set-cookie, euconsent-v2, CQBihYAQBihYABcAKDFRA8EgAAAAAAAAAAqIIzQAARmgAAAA.yAMbVNg, , , reload, 1)`
`ileauxepices.com##+js(trusted-set-local-storage-item, 'sddan:cmp', '{"lp":1720590795015}')` <= $timestamp$
`ileauxepices.com##+js(trusted-set-local-storage-item, 'sddan:cmp:stc', CQBihYAQBihYABcAKDFRA8EgAAAAAAAAAAqIIzQAARmgAAAA.yAMbVNg)`
`ileauxepices.com##+js(trusted-set-local-storage-item, 'sddan:cmp:stc:ts', 1736142794974)` <= $timestamp$

This would definitely be useful for other sites/settings I think.

**Comment by stephenhawk8054** [comment]
There's `$now$` for epoch time.

Does this work?

```adb
ileauxepices.com##+js(trusted-set-local-storage-item, sddan:cmp, {"lp":$now$})
```

**Reply by K-mikaZ** [reply]
Oh sorry, I didn't see it in the docs.
Yes it works well.
Thanks again, and sorry for the inconvenience.
Do you perhaps have a link to the documentation of possible tags and their equivalence?

**Reply by stephenhawk8054** [reply]
Yeah, currently the wiki has not caught up with the code yet. You can see some date-related values here: https://github.com/gorhill/uBlock/blob/e3b71a586183b0a8b984d93bcd5be58ab109f462/assets/resources/scriptlets.js#L1065-L1075

**Reply by K-mikaZ** [reply]
Thanks again, I'll mark the topic as resolved.

**Reply by garry-ut99** [reply]
> K-mikaZ : Oh sorry, I didn't see it in the docs.
> stephenhawk8054 : Yeah, currently the wiki has not caught up with the code yet.  You can see some date-related values here:

That's not true, as `$now$` , `$currentDate$` and `$currentISODate$` are documented:

> K-mikaZ Do you perhaps have a link to the documentation of possible tags and their equivalence?

- https://github.com/gorhill/uBlock/wiki/Resources-Library#trusted-set-cookiejs-
- https://github.com/gorhill/uBlock/wiki/Resources-Library#trusted-set-local-storage-itemjs-
- https://github.com/gorhill/uBlock/wiki/Resources-Library#trusted-set-session-storage-itemjs-
- https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#trusted-set-cookiejs- (`$currentISODate$`)

**Reply by krystian3w** [reply]
The wiki has an edit history and you can view the old status by add into URL `/commit`.

Theoretically, for newbies, `Date.now()` and `(new Date).getTime()` may be incomprehensible in section `trusted-slsi` - Maybe sample dates should be in all bullet points list.

----

Additionally, it is worth noting that page as uAssets issue, added to the Cookie Notices list as the author states the need to accept the GDPR/cookies policy and doesn't exist trick with low permission scriptlets.
